extended/src/main/java/io/kubernetes/client/extended/event/legacy/LegacyEventBroadcaster.java

```diff
@@ -0,0 +1,161 @@
+                event.getMetadata().getName(),
+                event.getMetadata().getNamespace(),
+                event,
+                null,
+                null,
+                null);
+          }
+
+          @Override
+          public V1Event patch(V1Event event, V1Patch patch) throws ApiException {
+            return PatchUtils.patch(
+                V1Event.class,
+                () ->
+                    coreV1Api.patchNamespacedEventCall(
+                        event.getMetadata().getName(),
+                        event.getMetadata().getNamespace(),
+                        patch,
+                        null,
+                        null,
+                        null,
+                        null,
+                        null),
+                V1Patch.PATCH_FORMAT_STRATEGIC_MERGE_PATCH,
+                coreV1Api.getApiClient());
+          }
+        });
+  }
+
+  public LegacyEventBroadcaster(EventCorrelator correlator, EventSink sink) {
+    this.pendingEventQueue = new LinkedBlockingQueue<>();
+    this.eventProcessingWorker = Executors.newSingleThreadExecutor();
+    this.eventCorrelator = correlator;
+    this.sleepDuration = Duration.ofSeconds(10);
+    this.eventSink = sink;
+  }
+
+  private BlockingQueue<V1Event> pendingEventQueue;
+  private ExecutorService eventProcessingWorker;
+  private EventSink eventSink;
+  private EventCorrelator eventCorrelator;
+  private Duration sleepDuration;
+  private boolean shuttingDown;
+
+  @Override
+  public EventRecorder newRecorder(V1EventSource eventSource) {
+    return new ObjectReferenceResolvingEventRecorder(this.pendingEventQueue);
+  }
+
+  @Override
+  public void startRecording() {
+    this.eventProcessingWorker.submit(
+        () -> {
+          while (true) {
+            try {
+              V1Event event = pendingEventQueue.poll(100, TimeUnit.MILLISECONDS);
+              if (event != null) {
+                recordToSink(event);
+              }
+            } catch (InterruptedException e) {
+              logger.info("shutdown signaled");
+            }
+            if (this.shuttingDown) {
+              return;
+            }
+          }
+        });
+  }
+
+  @Override
+  public void shutdown() {
+    this.eventProcessingWorker.shutdown();
+    this.shuttingDown = true;
+  }
+
+  private void recordToSink(V1Event event) {
+    Optional<MutablePair<V1Event, V1Patch>> eventAndPatch = this.eventCorrelator.correlate(event);
+    if (!eventAndPatch.isPresent()) {
+      // skip
+      return;
+    }
+    V1Event recordingEvent = eventAndPatch.get().getLeft();
+    V1Patch patch = eventAndPatch.get().getRight();
+    final int maxTriesPerEvent = 10;
+    for (int retries = 0; retries < maxTriesPerEvent; retries++) {
+      if (recordEvent(recordingEvent, patch, event.getCount() > 1)) {
+        break;
+      }
+      try {
+        Thread.sleep(sleepDuration.toMillis());
+      } catch (InterruptedException e) {
+        return;
```

Comment:
is this needed?

Reply:
mm the outter try-catch handle the interruptedexception, removed